### ISSUE TITLE
[Feature]:  Divide CanvasX from FabricJS code

### ISSUE THREAD

**Issue by asturur** [boardx/canvasx#4]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

Working with CanvasX as a fork of FabricJS is confusing.
Ideally CanvasX does not changes to fabricJS and imports the functions.
If some function needs to be customized heavily then it can be copied

### Current State

-

**Comment by usamshen** [comment]
@asturur

I agree, let's import the Fabric.js library into the project and remove all the source code of Fabric.js. I am willing to implement this change.

I have two questions regarding this:

1. What is the best option for adding extra fields or functions to canvas and fabricObject?
2. Can we ensure that all core types, schemas, and interfaces of Fabric.js are exported? This way, when we customize a class, we won't need to copy the types or interfaces.

Thank you!
